exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MetricAdapter.java

```diff
@@ -60,11 +60,15 @@
           entryResource.getValue().entrySet()) {
         instrumentationLibraryMetrics.add(buildInstrumentationLibraryMetrics(entryLibrary));
       }
-      resourceMetrics.add(
+      ResourceMetrics.Builder resourceMetricsBuilder =
           ResourceMetrics.newBuilder()
               .setResource(ResourceAdapter.toProtoResource(entryResource.getKey()))
-              .addAllInstrumentationLibraryMetrics(instrumentationLibraryMetrics)
-              .build());
+              .addAllInstrumentationLibraryMetrics(instrumentationLibraryMetrics);
+      String schemaUrl = entryResource.getKey().getSchemaUrl();
+      if (schemaUrl != null) {
```

Comment:
I'd extract the builder creation to a separate helper method to keep the fluent builder flow.